AppController/djinn.rb

```diff
@@ -665,35 +665,13 @@ def relocate_app(appid, http_port, https_port, secret)
          "#{Nginx::END_PORT - Nginx::SSL_PORT_OFFSET}."
     end
 
-    # Next, make sure that no other app is using either of these ports for
-    # nginx, haproxy, or the AppServer itself.
-    @app_info_map.each { |app, info|
-      # Of course, it's fine if it's our app on a given port, since we're
-      # relocating that app.
-      next if app == appid
-
-      if [http_port, https_port].include?(info['nginx'])
-        return "Error: Port in use by nginx for app #{app}"
-      end
-
-      if [http_port, https_port].include?(info['nginx_https'])
-        return "Error: Port in use by nginx for app #{app}"
-      end
-
-      if [http_port, https_port].include?(info['haproxy'])
-        return "Error: Port in use by haproxy for app #{app}"
-      end
+    if find_lowest_free_port(http_port, http_port, appid) == -1
```

Comment:
Yes it is correct: we are not looking for a range here, but if that specific port is available.